util/shell/lex_test.go

```diff
@@ -48,6 +48,26 @@ func TestShellParserMandatoryEnvVars(t *testing.T) {
 	require.Contains(t, err.Error(), "message herex")
 }
 
+func TestShellParserHash(t *testing.T) {
```

Comment:
can you create a test for the case where the replacement doesn't occur?

e.g. `"${foo#greetings}"` -> `hello world`

Reply:
Sure, sounds good.